File: backend/logistics-shipments-service/src/main/java/com/tdtu/logistics_shipments_service/dto/request/CreateShipmentRequest.java
package com.tdtu.logistics_shipments_service.dto.request;

import com.tdtu.logistics_shipments_service.enumrator.ShipmentStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@Builder
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
@NoArgsConstructor
@AllArgsConstructor
public class CreateShipmentRequest {
	String trackingNumber;
	Long shipper;
	String shipmentMethod;
	Long fromWarehouseId;
	Long toWarehouseId;
	ShipmentStatus shipmentStatus;
	String departureTime;
	List<String> orders;
	List<CreateShipmentSegmentRequest> shipmentSegmentRequests;
}

File: my-app/src/modules/shipment/dto/request/CreateShipmentRequest.ts
import ShipmentStatus from "../../models/ShipmentStatus";
import CreateShipmentSegmentRequest from "../request/CreateShipmentSegmentRequest"

interface CreateShipmentRequest {
    trackingNumber: string;
    shipper: string;
    shipmentMethod: string;
    fromWarehouseId: number;
    toWarehouseId: number;
    shipmentStatus: ShipmentStatus;
    departureTime: string
    orders: string[];
    shipmentSegmentRequests: CreateShipmentSegmentRequest[],
    // vehicle: {
    //     id: number,
    //     name: string,
    //     employee: { id: number, name: string, role: string }
    // }
}

export type { CreateShipmentRequest };
